### PR TITLE
ci: add checks for `iptables-nft` in integration test runner

### PR DESCRIPTION
**Pull request by thevar1able**
### Changelog category (leave one):
- CI Fix or Improvement (changelog entry is not required)

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit df7d44bc3adf2fecbd8506e155d0016a9847f7f3 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/68902/df7d44bc3adf2fecbd8506e155d0016a9847f7f3/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Builds</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/68752/3cd11afa61545e366c003339886e34d1427a508b/builds/report.html">✅ success</a></td></tr>
<tr><td>ClickBench</td><td>Runs [ClickBench](https://github.com/ClickHouse/ClickBench/) with instant-attach table</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/1cdccd527f0cbf5629b21d29970e28d5156003dc/clickbench__aarch64_.html">✅ success</a></td></tr>
<tr><td>Compatibility check</td><td>Checks that <b>clickhouse</b> binary runs on distributions with old libc versions. If it fails, ask a maintainer for help</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/1cdccd527f0cbf5629b21d29970e28d5156003dc/compatibility_check__aarch64_.html">✅ success</a></td></tr>
<tr><td>Docker keeper image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/1cdccd527f0cbf5629b21d29970e28d5156003dc/docker_keeper_image.html">✅ success</a></td></tr>
<tr><td>Docker server image</td><td>The check to build and optionally push the mentioned image to docker hub</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/1cdccd527f0cbf5629b21d29970e28d5156003dc/docker_server_image.html">✅ success</a></td></tr>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/1cdccd527f0cbf5629b21d29970e28d5156003dc/docs_check.html">✅ success</a></td></tr>
<tr><td>Fast test</td><td>Normally this is the first check that is ran for a PR. It builds ClickHouse and runs most of <a href="https://clickhouse.com/docs/en/development/tests#functional-tests">stateless functional tests</a>, omitting some. If it fails, further checks are not started until it is fixed. Look at the report to see which tests fail, then reproduce the failure locally as described <a href="https://clickhouse.com/docs/en/development/tests#functional-test-locally">here</a></td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/1cdccd527f0cbf5629b21d29970e28d5156003dc/fast_test.html">✅ success</a></td></tr>
<tr><td>Install packages</td><td>Checks that the built packages are installable in a clear environment</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/1cdccd527f0cbf5629b21d29970e28d5156003dc/install_packages__aarch64_.html">✅ success</a></td></tr>
<tr><td>Stateful tests</td><td>Runs stateful functional tests for ClickHouse binaries built in various configurations -- release, debug, with sanitizers, etc</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/1cdccd527f0cbf5629b21d29970e28d5156003dc/stateful_tests__aarch64_.html">✅ success</a></td></tr>
<tr><td>Stateless tests</td><td>Runs stateless functional tests for ClickHouse binaries built in various configurations -- release, debug, with sanitizers, etc</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/1cdccd527f0cbf5629b21d29970e28d5156003dc/stateless_tests__aarch64_.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/1cdccd527f0cbf5629b21d29970e28d5156003dc/unit_tests__msan_.html">✅ success</a></td></tr>
<tr><td>Upgrade check</td><td>Runs stress tests on server version from last release and then tries to upgrade it to the version from the PR. It checks if the new server can successfully startup without any errors, crashes or sanitizer asserts</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/68752/3cd11afa61545e366c003339886e34d1427a508b/upgrade_check__debug_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: tests/integration/runner [hunk 1]
```diff
@@ -135,6 +136,43 @@ def check_args_and_update_paths(args):
...
+
+        if not ip6tables_path:
+            return
+
+        file_info = os.stat(ip6tables_path)
+        file_info_str = str(file_info)
+
+        if "legacy" in file_info_str:
+            print(
+                "ip6tables is in 'legacy' mode. This is not supported. Please switch to 'nftables' mode."
```
Comment (qoega) [inline tests/integration/runner:167]: Can we provide command to use or at least a link to learn how to switch?

Comment (thevar1able) [reply tests/integration/runner:167]: I can add `consult your distribution manual`, will this be sufficient for you?

**Review by Felixoid — APPROVED** [review]
Looks good to me, although https://github.com/search?q=head%3Async-upstream%2Fpr%2F68902+org%3AClickHouse+type%3Apr&type=pullrequests didn't report the success for some reason.

@maxknv, how it works now? On which stage?

**Comment by maxknv** [comment]
LGTM

> Looks good to me, although https://github.com/search?q=head%3Async-upstream%2Fpr%2F68902+org%3AClickHouse+type%3Apr&type=pullrequests didn't report the success for some reason.
> 
> @maxknv, how it works now? On which stage?

it seems Cloud fork sync status gets rewritten by sync script

File: tests/integration/runner (hunk 1 above)
Comment (qoega) [reply tests/integration/runner]: Actually not as we run docker in docker. You better say that this should be configured on host system... 
It was not trivial and I'm not sure I made it correctly:
https://www.danami.com/clients/knowledgebase/194/How-can-I-switch-to-iptables-nft-from-nftables-after-upgrading-Debian-or-Ubuntu.html?language=english

Comment (thevar1able) [reply tests/integration/runner]: :ok_hand:

**Pull request merged**